cmd/controller.go

```diff
@@ -1,60 +1,62 @@
-package command
+package cmd
 
 import (
 	xkube "github.com/crossplane-contrib/provider-kubernetes/apis/object/v1alpha1"
 	"github.com/go-logr/logr"
 	"github.com/spf13/cobra"
-	"github.com/vshn/appcat-apiserver/controller/postgres"
-	vshnv1 "github.com/vshn/component-appcat/apis/vshn/v1"
+	vshnv1 "github.com/vshn/appcat-apiserver/apis/vshn/v1"
+	"github.com/vshn/appcat-apiserver/pkg/controller/postgres"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/healthz"
 )
 
-func init() {
-	_ = corev1.SchemeBuilder.AddToScheme(s)
-	_ = xkube.SchemeBuilder.AddToScheme(s)
-	_ = vshnv1.SchemeBuilder.SchemeBuilder.AddToScheme(s)
-
-	controller.Flags().StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
-	controller.Flags().StringVar(&healthAddr, "health-addr", ":8081", "The address the probe endpoint binds to.")
-	controller.Flags().BoolVar(&leaderElect, "leader-elect", false, "Enable leader election for controller manager. "+
-		"Enabling this will ensure there is only one active controller manager.")
-}
-
-var (
+type controller struct {
+	scheme                  *runtime.Scheme
 	metricsAddr, healthAddr string
 	leaderElect             bool
-)
+}
 
-var controller = &cobra.Command{
+var c = controller{
+	scheme: runtime.NewScheme(),
+}
+
+var ControllerCMD = &cobra.Command{
 	Use:   "controller",
-	Short: "Postgres Controller",
-	Long:  "Run the Postgres Controller",
-	RunE:  executeController,
+	Short: "Controller",
+	Long:  "Run the Controller",
+	RunE:  c.executeController,
 }
 
-var s = runtime.NewScheme()
+func init() {
+	_ = corev1.SchemeBuilder.AddToScheme(c.scheme)
```

Comment:
We register the same schemes in `sliexporter.go` as well. Maybe put the register calls into a function that we call in `init()`? Then we won't forget to add new CRDs that easily.